### PR TITLE
De-dup ledgers - db_ledger is now the only ledger written to disk

### PR DESCRIPTION
**Pull request by mvines**
`src/ledger_write_stage.rs` is now gone, and bank/fullnode boot from db_ledger.

TODO:
* [x] Unbreak storage_stage
* [x] Fix multinode tests
* [x] Might need to adjust scripts as ledger-tool can't open db_ledger while the fullnode is running
* [x] ...
* [x] Upon r+, file a new issue to track updating replicators to use db_ledger

### CODE REVIEW

File: src/db_ledger.rs [hunk 1]
```diff
@@ -626,6 +629,15 @@ impl DbLedger {
         Ok(EntryIterator { db_iterator })
     }
 
+    pub fn get_entries_bytes(
+        &self,
+        _start_index: u64,
+        _num_entries: u64,
+        _buf: &mut [u8],
+    ) -> io::Result<(u64, u64)> {
+        Err(io::Error::new(io::ErrorKind::Other, "TODO"))
```
Comment (mvines) [inline src/db_ledger.rs]: Placeholder for plumbing db_ledger into `src/chacha_cuda.rs`

**Comment by mvines** [comment]
`src/replicator.rs` no longer works with this PR, as the ledger `data` file is no longer written so `chacha_cbc_encrypt_file()` fails.  This'll be part of the follow-up issue I'll file

File: src/fullnode.rs [hunk 2]
```diff
@@ -202,6 +200,12 @@ impl Fullnode {
         let exit = Arc::new(AtomicBool::new(false));
         let bank = Arc::new(bank);
 
+        let db_ledger = if db_ledger.is_none() {
```
Comment (garious) [inline src/fullnode.rs]: Is Clippy running over this project? It should have suggested:

```rust
let db_ledger = db_ledger.unwrap_or_else(|| Self::make_db_ledger(ledger_path));
```

Comment (mvines) [reply src/fullnode.rs]: Yep, it is.  When I remove `#[allow(clippy::too_many_arguments)]` in the same file clippy complains.   I've applied this change.

File: src/db_ledger.rs [hunk 3]
```diff
@@ -329,6 +331,7 @@ impl DbLedger {
     }
 
     pub fn destroy(ledger_path: &str) -> Result<()> {
+        create_dir_all(&ledger_path)?;
```
Comment (rob-solana) [inline src/db_ledger.rs]: this is weird...

Comment (mvines) [reply src/db_ledger.rs]: yeah it deserves a comment, added.  DB::Destroy fails if the parent directory doesn't exist 🙃

**Pull request merged**